### PR TITLE
return an error response when attempting to build a demo whose html is invalid.

### PR DESCRIPTION
**Pull request by JakeChampion**
If a component is published and it's demo contains incorrect html we have decided that the service should respond with an error message stating the html is incorrect/invalid because we know some of our users copy/paste the demo html and we don’t want to be responsible for copy-pasted html errors within our users' products.

### CODE REVIEW

File: lib/middleware/v3/outputDemo.js [hunk 1]
```diff
@@ -37,7 +37,13 @@ const rimrafOptions = {
 };
 
 const extractMinimalHtml = (fullHtml) => {
-	const $ = cheerio.load(fullHtml);
+	let formattedHTML;
+	try {
+		formattedHTML = prettier.format(fullHtml, { parser: 'html' });
+	} catch (error) {
+		throw new ComponentError('The HTML in the demo contains syntax errors.');
```
Comment (JakeChampion) [inline lib/middleware/v3/outputDemo.js:44]: @Financial-Times/origami-core, any suggestions on how we could improve this error message for when the demo html is syntactically invalid?

Comment (chee) [reply lib/middleware/v3/outputDemo.js:44]: Does the error itself have information about where the parsing errors took place? can we print those to give them idea of where to look?

Comment (JakeChampion) [reply lib/middleware/v3/outputDemo.js:44]: Yes, it has it inside a very big string which contains the entire html document. I'll make it return that 👍

Comment (JakeChampion) [reply lib/middleware/v3/outputDemo.js]: Done in [`f34696f` (#479)](https://github.com/Financial-Times/origami-build-service/pull/479/commits/f34696fd1f46e895c73f74ba6fefec65ca1c8b71)

**Review by chee — APPROVED** [review]

**Pull request merged**